### ISSUE TITLE
Application start fails when marking more than one Command with @Transactional

### ISSUE THREAD

**Issue by ydautremay** [seedstack/seed#191]
Hi team

I created several commands (org.seedstack.seed.spi.command.Command) to be used in shell. As these command are supposed to query repositories I annotated them with @Transactional. The problem is when more than one command is annotated Guice explodes at startup with the following error :
```
org.seedstack.seed.SeedException: (CORE) Unexpected exception

Causes
------
1. com.google.inject.CreationException: Guice creation errors:

1) Unable to method intercept: org.ydautremay.ouist.commands.PlayCommand
  at org.seedstack.seed.core.internal.command.impl.CommandModule.configure(CommandModule.java:31)

1 error
2. java.lang.ArrayIndexOutOfBoundsException: 7620

Stacktrace
----------
	at org.seedstack.seed.SeedException.wrap(SeedException.java:438)
	at org.seedstack.seed.core.SeedMain.handleException(SeedMain.java:83)
	at org.seedstack.seed.core.SeedMain.main(SeedMain.java:58)
Caused by: com.google.inject.CreationException: Guice creation errors:

1) Unable to method intercept: org.ydautremay.ouist.commands.PlayCommand
  at org.seedstack.seed.core.internal.command.impl.CommandModule.configure(CommandModule.java:31)

1 error
	at com.google.inject.internal.Errors.throwCreationExceptionIfErrorsExist(Errors.java:448)
	at com.google.inject.internal.InternalInjectorCreator.initializeStatically(InternalInjectorCreator.java:155)
	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:107)
	at com.google.inject.Guice.createInjector(Guice.java:96)
	at com.google.inject.Guice.createInjector(Guice.java:84)
	at io.nuun.kernel.core.internal.KernelCore.start(KernelCore.java:273)
	at org.seedstack.seed.cli.SeedRunner.startKernel(SeedRunner.java:178)
	at org.seedstack.seed.cli.SeedRunner.execute(SeedRunner.java:101)
	at org.seedstack.seed.cli.SeedRunner.launch(SeedRunner.java:55)
	at org.seedstack.seed.core.SeedMain.main(SeedMain.java:56)
Caused by: java.lang.ArrayIndexOutOfBoundsException: 7620
	at com.google.inject.internal.asm.$ClassReader.<init>(Unknown Source)
	at com.google.inject.internal.asm.$ClassReader.<init>(Unknown Source)
	at com.google.inject.internal.asm.$ClassReader.<init>(Unknown Source)
	at com.google.inject.internal.asm.$ClassReader.<init>(Unknown Source)
	at com.google.inject.internal.cglib.proxy.$BridgeMethodResolver.resolveAll(BridgeMethodResolver.java:61)
	at com.google.inject.internal.cglib.proxy.$Enhancer.emitMethods(Enhancer.java:911)
	at com.google.inject.internal.cglib.proxy.$Enhancer.generateClass(Enhancer.java:498)
	at com.google.inject.internal.cglib.core.$DefaultGeneratorStrategy.generate(DefaultGeneratorStrategy.java:25)
	at com.google.inject.internal.cglib.core.$AbstractClassGenerator.create(AbstractClassGenerator.java:216)
	at com.google.inject.internal.cglib.proxy.$Enhancer.createHelper(Enhancer.java:377)
	at com.google.inject.internal.cglib.proxy.$Enhancer.createClass(Enhancer.java:317)
	at com.google.inject.internal.ProxyFactory$ProxyConstructor.<init>(ProxyFactory.java:249)
	at com.google.inject.internal.ProxyFactory.create(ProxyFactory.java:175)
	at com.google.inject.internal.ConstructorInjectorStore.createConstructor(ConstructorInjectorStore.java:90)
	at com.google.inject.internal.ConstructorInjectorStore.access$000(ConstructorInjectorStore.java:29)
	at com.google.inject.internal.ConstructorInjectorStore$1.create(ConstructorInjectorStore.java:37)
	at com.google.inject.internal.ConstructorInjectorStore$1.create(ConstructorInjectorStore.java:33)
	at com.google.inject.internal.FailableCache$1.apply(FailableCache.java:38)
	at com.google.common.collect.ComputingConcurrentHashMap$ComputingValueReference.compute(ComputingConcurrentHashMap.java:358)
	at com.google.common.collect.ComputingConcurrentHashMap$ComputingSegment.compute(ComputingConcurrentHashMap.java:184)
	at com.google.common.collect.ComputingConcurrentHashMap$ComputingSegment.getOrCompute(ComputingConcurrentHashMap.java:153)
	at com.google.common.collect.ComputingConcurrentHashMap.getOrCompute(ComputingConcurrentHashMap.java:69)
	at com.google.common.collect.ComputingConcurrentHashMap$ComputingMapAdapter.get(ComputingConcurrentHashMap.java:396)
	at com.google.inject.internal.FailableCache.get(FailableCache.java:49)
	at com.google.inject.internal.ConstructorInjectorStore.get(ConstructorInjectorStore.java:50)
	at com.google.inject.internal.ConstructorBindingImpl.initialize(ConstructorBindingImpl.java:137)
	at com.google.inject.internal.InjectorImpl.initializeJitBinding(InjectorImpl.java:543)
	at com.google.inject.internal.InjectorImpl.createJustInTimeBinding(InjectorImpl.java:865)
	at com.google.inject.internal.InjectorImpl.createJustInTimeBindingRecursive(InjectorImpl.java:790)
	at com.google.inject.internal.InjectorImpl.getJustInTimeBinding(InjectorImpl.java:278)
	at com.google.inject.internal.InjectorImpl.getBindingOrThrow(InjectorImpl.java:210)
	at com.google.inject.internal.InjectorImpl.getInternalFactory(InjectorImpl.java:871)
	at com.google.inject.internal.FactoryProxy.notify(FactoryProxy.java:46)
	at com.google.inject.internal.ProcessedBindingData.runCreationListeners(ProcessedBindingData.java:50)
	at com.google.inject.internal.InternalInjectorCreator.initializeStatically(InternalInjectorCreator.java:134)
	... 8 more
```

For now I'll encapsulate my calls to the repositories in service methods but I thought you'd like to know.

**Comment by adrienlauer** [comment]
It seems that this kind of error is related to some incompatibility between Guice 3 and Java 8 (see google/guice#757). As `Command` is a single-method interface, it can get used with the lambda syntax, perhaps triggering the errror.

I started a work to move to Guice 4.0, starting with nuun (nuun-io/kernel#56) and followed by seed (https://github.com/adrienlauer/seed/tree/nuun-1.0). Sadly, for Seed, I had to patch the Guice integration of Shiro, as it is not compatible with Guice 4.0.

**Comment by ydautremay** [comment]
Nice one ! There was actually lambda expressions in my commands. When I remove them the application starts fine.

**Comment by adrienlauer** [comment]
Fixed by #196

**Comment by adrienlauer** [comment]
Version 2.3.0 is estimated for late april 2016.
